app/src/main/java/com/github/sdp_begreen/begreen/fragments/MapFragment.kt

```diff
@@ -123,4 +149,66 @@ class MapFragment : Fragment() {
             Toast.makeText(requireActivity(), getString(R.string.user_current_location_error), Toast.LENGTH_LONG).show()
         }
     }
+
+    /**
+     * Helper functions that displays bins markers on the map
+     */
+    private fun displayBinsMarkers(bins: Set<Bin>) {
+
+        // Clear old bins so we don't display removed or duplicates bins
+        map.clear()
+
+        for (bin in bins) {
+
+            val location = LatLng(bin.lat, bin.long)
+
+            val marker = map.addMarker(
+                MarkerOptions()
+                    .position(location)
+                    .title(bin.type.toString())
+                    .icon(BitmapDescriptorFactory.defaultMarker(bin.type.markerColor))
+            )
+
+            marker!!.tag = bin.id
+        }
+
+        // Setup the markers click listener action
+        setupMarkersClick()
+    }
+
+    /**
+     * Helper function that setups a marker click listener action
+     */
+    private fun setupMarkersClick() {
+
+        // Delete a bin when the user clicks on the associated marker
+        map.setOnMarkerClickListener { marker ->
+
+            // Remove the bin associated to this marker
+            BinsFakeDatabase.removeBin(marker.tag as Int)
+            displayBinsMarkers(BinsFakeDatabase.fakeBins)
+
+            true // Return true to indicate that the event has been handled
+        }
+    }
+
+    /**
+     * Helper function that add a bin marker to the user current location
+     */
+    private fun addNewBin() {
+
+        if (ContextCompat.checkSelfPermission(
+                requireContext(),
+                Manifest.permission.ACCESS_FINE_LOCATION
+            ) != PackageManager.PERMISSION_GRANTED) {
+
+            requestPermissionLauncher.launch(Manifest.permission.ACCESS_FINE_LOCATION)
+
+        } else {
+
+            val bin = Bin(0, BinType.PLASTIC, userLocation!!.latitude, userLocation!!.latitude)
```

Comment:
```suggestion
            val bin = Bin(0, BinType.PLASTIC, userLocation!!.latitude, userLocation!!.longitude)
```
I think there's a little typo here